### DISCUSSION TITLE
关于HTTP发送参数配置未生效问题

### DISCUSSION THREAD

**Discussion by kuman62** [apache/dolphinscheduler#15322] [Q&A]
基于3.2.0版本
任务实例执行日志如下：
[INFO] 2023-12-14 23:36:00.100 +0800 - ***********************************************************************************************
[INFO] 2023-12-14 23:36:00.100 +0800 - *********************************  Load task instance plugin  *********************************
[INFO] 2023-12-14 23:36:00.100 +0800 - ***********************************************************************************************
[INFO] 2023-12-14 23:36:00.101 +0800 - Send task status RUNNING_EXECUTION master: 172.21.0.1:1234
[WARN] 2023-12-14 23:36:00.101 +0800 - Current tenant is default tenant, will use paas to execute the task
[INFO] 2023-12-14 23:36:00.101 +0800 - TenantCode: default check successfully
[INFO] 2023-12-14 23:36:00.101 +0800 - WorkflowInstanceExecDir: /tmp/dolphinscheduler/exec/process/default/11730447389472/11730497764386_9/12842/12842 check successfully
[INFO] 2023-12-14 23:36:00.101 +0800 - Download resources: {} successfully
[INFO] 2023-12-14 23:36:00.102 +0800 - Download upstream files: [] successfully
[INFO] 2023-12-14 23:36:00.102 +0800 - Task plugin instance: HTTP create successfully
[INFO] 2023-12-14 23:36:00.102 +0800 - Initialize http task params {
  "localParams" : [ ],
  "varPool" : null,
  "url" : "http://127.0.0.1:9086/sendMsg",
  "httpMethod" : "POST",
  "httpParams" : [ {
    "prop" : "request",
    "httpParametersType" : "BODY",
    "value" : "{\"eventCode\":\"EVENT10001\",\"messageKeys\":\"1000#1000#1000\",\"messageTags\":\"unionCollect\",\"msgBody\":\"hello world\"}"
  }, {
    "prop" : "Content-Type",
    "httpParametersType" : "HEADERS",
    "value" : "application/json;charset=UTF-8"
  } ],
  "httpCheckCondition" : "STATUS_CODE_DEFAULT",
  "condition" : "",
  "connectTimeout" : 60000,
  "socketTimeout" : 60000
}
[INFO] 2023-12-14 23:36:00.102 +0800 - Success initialized task plugin instance successfully
[INFO] 2023-12-14 23:36:00.102 +0800 - Set taskVarPool: null successfully
[INFO] 2023-12-14 23:36:00.102 +0800 - ***********************************************************************************************
[INFO] 2023-12-14 23:36:00.102 +0800 - *********************************  Execute task instance  *************************************
[INFO] 2023-12-14 23:36:00.102 +0800 - ***********************************************************************************************
[INFO] 2023-12-14 23:36:00.104 +0800 - http request params：{"prop":"request","httpParametersType":"BODY","value":"{\"eventCode\":\"EVENT10001\",\"messageKeys\":\"1000#1000#1000\",\"messageTags\":\"unionCollect\",\"msgBody\":\"hello world\"}"}
[INFO] 2023-12-14 23:36:00.104 +0800 - http request params：{"prop":"Content-Type","httpParametersType":"HEADERS","value":"application/json;charset=UTF-8"}
[INFO] 2023-12-14 23:36:00.112 +0800 - startTime: 2023-12-14 23:36:00, httpUrl: http://127.0.0.1:9086/sendMsg, httpMethod: POST, costTime : 10 milliseconds, statusCode : 200, body : 发送失败, log : null
[INFO] 2023-12-14 23:36:00.113 +0800 - ***********************************************************************************************
[INFO] 2023-12-14 23:36:00.113 +0800 - *********************************  Finalize task instance  ************************************

任务流定义
![无标题](https://github.com/apache/dolphinscheduler/assets/129487829/764d3240-0664-4e8c-8e07-de64d9775cf2)

关于prop配置说明在官方文档里面找到相关说明，建议官方对HTTP类型任务，增加POST，json报文的配置样例

业务服务端（http://127.0.0.1:9086/sendMsg）日志显示，包体消息为空，导致业务处理失败。

**Comment by peak-xu** [comment]
解决了吗我也有同样的问题

**Comment by 966699** [comment]
同问，我也遇到了使用http任务时传入参数正确，但是就是感觉没生效的问题
